lib/spack/spack/compilers/clang.py

```diff
@@ -121,6 +122,28 @@ def _find_full_path(self, path):
         full_path = xcrun('-f', basename, output=str)
         return full_path.strip()
 
+    @classmethod
+    def fc_version(cls, fc):
+        version = get_compiler_version(
+            fc, '-dumpversion',
+            # older gfortran versions don't have simple dumpversion output.
+            r'(?:GNU Fortran \(GCC\))?(\d+\.\d+(?:\.\d+)?)')
+        # This is horribly ad hoc, we need to map from gcc/gfortran version
+        # to clang version, but there could be multiple clang
+        # versions that work for a single gcc/gfortran version
```

Comment:
it's probably even anything to anything (I never had to worry which version of clang i use with what version of gfortran). But i don't have a better solution to this problem either.